CMakeLists.txt

```diff
@@ -1,235 +1,143 @@
+    PUBLIC
+        Eigen3::Eigen
 )
 
-## Mark other files for installation (e.g. launch and bag files, etc.)
-install(DIRECTORY
-   config
-   launch
-   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+include(GNUInstallDirs)
+install(TARGETS ${PROJECT_NAME}
+    EXPORT ${PROJECT_NAME}Targets
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    ${PROJECT_NAME}ConfigVersion.cmake
+    COMPATIBILITY AnyNewerVersion
+)
+install(DIRECTORY include/${PROJECT_NAME} DESTINATION include)
+install(EXPORT ${PROJECT_NAME}Targets
+    FILE "${PROJECT_NAME}Targets.cmake"
+    NAMESPACE steering_functions::
+    DESTINATION "lib/cmake/${PROJECT_NAME}"
+)
+install(
+    FILES
+        "${PROJECT_NAME}Config.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    DESTINATION "lib/cmake/${PROJECT_NAME}"
 )
 
-#############
-## Testing ##
-#############
-
-if(CATKIN_ENABLE_TESTING)
-  ## Add gtest based cpp test target and link libraries
-  catkin_add_gtest(steering_functions_test test/steering_functions_test.cpp)
-  if(TARGET steering_functions_test)
-    target_link_libraries(steering_functions_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-  endif()
-
-  catkin_add_gtest(fresnel_test test/fresnel_test.cpp)
-  if(TARGET fresnel_test)
-    target_link_libraries(fresnel_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-  endif()
-
-  catkin_add_gtest(timing_test test/timing_test.cpp)
-  if(TARGET timing_test)
-    target_link_libraries(timing_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-  endif()
-
-  catkin_add_gtest(jacobian_test test/jacobian_test.cpp)
-  if(TARGET jacobian_test)
-    target_link_libraries(jacobian_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-  endif()
-
-  catkin_add_gtest(hc_cc_circle_test test/hc_cc_circle_test.cpp)
-  if(TARGET hc_cc_circle_test)
-    target_link_libraries(hc_cc_circle_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-  endif()
+if(BUILD_WITH_ROS)
+    add_executable(${PROJECT_NAME}_node src/${PROJECT_NAME}_node.cpp)
+
+    target_link_libraries(${PROJECT_NAME}_node
+        PRIVATE
+            steering_functions
+            Eigen3::Eigen
+            ${catkin_LIBRARIES}
+    )
+
+    target_include_directories(${PROJECT_NAME}_node
+        PRIVATE
+            ${catkin_INCLUDE_DIRS}
+    )
+
+    install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_node
+        ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+        LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+        RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+    )
+
+    install(DIRECTORY include/${PROJECT_NAME}/
+        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+    )
+
+    install(DIRECTORY
+        config
+        launch
+        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+    )
+
```

Comment:
Any reason we need to line breaks here?

Reply:
Do you ask why I have:
```
target_link_libraries(${PROJECT_NAME}_node
        PRIVATE
            steering_functions
            Eigen3::Eigen
            ${catkin_LIBRARIES}
    )
```
instead of simply like this?
```
target_link_libraries(${PROJECT_NAME} PRIVATE steering_functions Eigen3::Eigen ${catkin_LIBRARIES})
```

This is a CMake convetion so it's easier to differentiate when you have both public and private linking of other libraries. Like in this example:
```
target_link_libraries(${PROJECT_NAME}_node
        PUBLIC
            libraryA
        PRIVATE
            libraryB
            libraryC
    )
```

Reply:
Just saying that the empty line in line 110 can be removed.